File: plugins/LayoutController/src/main/java/com/aliucord/plugins/layoutcontroller/patchers/NitroGiftButtonPatch.java
package com.aliucord.plugins.layoutcontroller.patchers;

import android.view.View;

import com.aliucord.plugins.layoutcontroller.patchers.base.BasePatcher;
import com.aliucord.plugins.layoutcontroller.util.Const;
import com.lytefast.flexinput.fragment.FlexInputFragment;
import com.lytefast.flexinput.fragment.FlexInputFragment$d;

import top.canyie.pine.Pine;

public class NitroGiftButtonPatch extends BasePatcher {

    public NitroGiftButtonPatch() throws Exception {
        super(Const.Key.GIFT_BUTTON_KEY, Const.Description.GIFT_BUTTON_DESCRIPTION, FlexInputFragment$d.class.getDeclaredMethod("invoke", Object.class));
    }

    @Override
    public void patchBody(Pine.CallFrame callFrame) {
        var fragment = (FlexInputFragment) ((FlexInputFragment$d) callFrame.thisObject).receiver;
        var binding = fragment.j();

        if (binding == null) return;

        binding.h.setVisibility(View.GONE);
        binding.m.setVisibility(View.GONE);
        binding.l.setVisibility(View.VISIBLE);
    }
}

File: plugins/LayoutController/src/main/java/com/aliucord/plugins/layoutcontroller/patchers/NitroGiftButtonPatch.kt
package com.aliucord.plugins.layoutcontroller.patchers

import android.view.View
import com.aliucord.plugins.layoutcontroller.patchers.base.BasePatcher
import com.aliucord.plugins.layoutcontroller.util.Description
import com.aliucord.plugins.layoutcontroller.util.Key
import com.lytefast.flexinput.fragment.FlexInputFragment
import com.lytefast.flexinput.fragment.`FlexInputFragment$d`
import de.robv.android.xposed.XC_MethodHook

class NitroGiftButtonPatch : BasePatcher(
    key = Key.GIFT_BUTTON_KEY,
    description = Description.GIFT_BUTTON_DESCRIPTION,
    classMember = `FlexInputFragment$d`::class.java.getDeclaredMethod(
        "invoke",
        Any::class.java
    )
) {
    override fun patchBody(callFrame: XC_MethodHook.MethodHookParam) {
        val fragment = (callFrame.thisObject as `FlexInputFragment$d`).receiver as FlexInputFragment
        val binding = fragment.j() ?: return
        binding.h.visibility = View.GONE
        binding.m.visibility = View.GONE
        binding.l.visibility = View.VISIBLE
    }
}